src/Cake/Commands/DebugCommand.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using Cake.Core.Scripting;
+using Cake.Scripting;
+
+namespace Cake.Commands
+{
+    /// <summary>
+    /// A command that builds and debugs a build script.
+    /// </summary>
+    internal sealed class DebugCommand : ICommand
+    {
+        private readonly IScriptRunner _scriptRunner;
+        private readonly DebugScriptHost _host;
+
+        // Delegate factory used by Autofac.
+        public delegate DebugCommand Factory();
+
+        public DebugCommand(IScriptRunner scriptRunner, DebugScriptHost host)
+        {
+            _scriptRunner = scriptRunner;
+            _host = host;
+        }
+
+        public bool Execute(CakeOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException("options");
+            }
+            _scriptRunner.Run(_host, options.Script, options.Arguments);
```

Comment:
@patriksvensson yes agree, but we should do wait in debug command execute and not RunTarget so fully enable debugging otherwise non task based code can't be debugged. i.e. arguments/global variables/etc.